# Spectrogram classification 모델 구현  
- 2차원 Spectrogram 데이터를 입력받아 위 모델과 동일한 역할을 수행하는 모델
- 기본 버전과 Skip-connection 버전으로 나누어 각각 진행
- 

## 데이터 처리와 분류

In [3]:
import numpy as np
import os

#데이터 저장 및 불러오기
data_path = os.getenv("HOME")+'/aiffel/speech_recognition/data/speech_wav_8000.npz'
speech_data = np.load(data_path)
print("✅")

✅


In [4]:
# npz파일로 이뤄진 데이터 저장 확인
print("Wave data shape : ", speech_data["wav_vals"].shape) 
print("Label data shape : ", speech_data["label_vals"].shape)
print("✅")

Wave data shape :  (50620, 8000)
Label data shape :  (50620, 1)
✅


데이터셋 1초길이 오디오 음성데이터 50620개  
데이터의 Sample rate는 16000개 이지만, 8000으로 re-sampling  
모두 1초의 길이를 가지는 오디오 음성데이터 - 각각 8000개의 sample data 있다

In [5]:
#데이터 확인
import IPython.display as ipd
import random

# 데이터 선택 (랜덤하게 선택하고 있으니, 여러번 실행해 보세요)
rand = random.randint(0, len(speech_data["wav_vals"]))
print("rand num : ", rand)

sr = 8000 # 1초동안 재생되는 샘플의 갯수
data = speech_data["wav_vals"][rand]
print("Wave data shape : ", data.shape)
print("label : ", speech_data["label_vals"][rand])

ipd.Audio(data, rate=sr)

rand num :  30821
Wave data shape :  (8000,)
label :  ['off']
